api/src/test/java/org/openmrs/api/OrderServiceTest.java

```diff
@@ -2969,4 +2969,101 @@ public void saveOrder_shouldPassIfAnKnownDrugOrderForTheSameDrugFormulationSpeci
 		orderService.saveOrder(order, orderContext);
 		assertNotNull(orderService.getOrder(order.getOrderId()));
 	}
+	
```

Comment:
As per acceptance criteria in OrderService.. Added tests  for following scenarios.
1) Adding new non coded drug - pass condition
2) Revising existing non coded drug - pass condition
3) discontinuing existing non coded drug - pass condition
4) Adding same non coded drug as existing - fail condition
5) Revising non existing non coded drug - fail condition
6) discontinuing non existing non coded drug - fail condition

Let us know if we miss out any acceptance criteria.